frontend/src/components/messages/ReplyRequestDialog.vue

```diff
@@ -97,28 +98,68 @@ export default {
     },
 
     /** 
-     * Create the reply and emit success event.
+     * Create the reply, update status to assigned, and emit success event.
      */
     async submit() {
-      this.$refs.replyRequestForm?.validate()
-      if (this.isFormComplete) {
+      if (this.isFormComplete && this.$refs.replyRequestForm?.validate()) {
         try {
           this.isLoading = true
-          const payload = {
-            assistanceRequestId: this.assistanceRequestId,
-            message: this.message,
-          }
-          await this.replyToAssistanceRequest(payload)
+          this.createReply(this.assistanceRequestId)
+          this.updateStatusToAssigned(this.assistanceRequestId)
+          this.updateStoredAssistanceRequest(this.assistanceRequestId)
           this.$emit('reply-success-event', true) // emit success to flag showing success message
         } catch (error) {
-          console.log(`Failed to create a reply for Assistance Request - ${error}`)
+          console.log(`Submit processing for reply failed  - ${error}`)
           throw error
         } finally {
           this.closeReplyRequestDialog()
           this.isLoading = false
         }
       }
     },
+
+    /**
+     * Create a reply for the assistance request.
+     */
+    async createReply(assistanceRequestId) {
```

Comment:
I probably wouldn't bother creating a method for these three as they are all essentially one-liners (if you don't bother creating a variable for payload). But if you feel it helps understanding/error handling then it's fine.

Reply:
I did it for the error handling, i flipped and flopped on it but eventually thought the error handling was worth it given 3 api calls in one method.